### PR TITLE
feat: update milvus model to support jina embeddings 3

### PR DESCRIPTION
**Pull request by bwanglzu**
This PR should allow `pymilvus` support `jina-embeddings-v3:

with 2 additional fields:

1. task_type: to properly load adapters
2. dimensions: allow user to truncate dimension with MRL.

### CODE REVIEW

File: milvus_model/dense/jinaai.py [hunk 1]
```diff
@@ -52,8 +55,13 @@ def __call__(self, texts: List[str]) -> List[np.array]:
         return self._call_jina_api(texts)
 
     def _call_jina_api(self, texts: List[str]):
+        data = {"input": texts, "model": self.model_name}
+        if self.model_name == "jina-embeddings-v3":
```
Comment (JoanFM) [inline milvus_model/dense/jinaai.py:59]: avoid doing this, just pass them if they are not None, the backend will return an error or not.

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Comment by wxywb** [comment]
Is there any doc about jina v3 embedding? what kind of tasks_type it can use?

**Comment by bwanglzu** [comment]
hi @wxywb yes i'm working on another PR to milvus-docs with detailed documentation, the tasks are:

```bash
+ **retrieval.query**: Used to encode user queries or questions in retrieval tasks.
+ **retrieval.passage**: Used to encode large documents in retrieval tasks at indexing time.
+ **classification**: Used to encode text for text classification tasks.
+ **text-matching**: Used to encode text for similarity matching, such as measuring similarity between two sentences.
+ **separation**: Used for clustering or reranking tasks.
```

**Comment by bwanglzu** [comment]
i need to figure out a good way for this PR, now my problem is, we have a `encode_query` and `encode_document` function built-in each embedding function, such as `JinaEmbeddingFunction`, i guess in these two functions, i need to use task_type=query and task_type=passage respectively.

While for other tasks, i'm not sure what is the best way to parse this `task_type`, so i overwrite the `__call__` function (not sure if it is desired).

Can you give me some suggestions?

Bo

**Comment by bwanglzu** [comment]
(btw if you are in the `jina_ai-x-milvus` joint slack channel, i can give you more detailed information :)

File: milvus_model/dense/jinaai.py [hunk 2]
```diff
@@ -43,17 +44,24 @@ def dim(self):
...
-        return self._call_jina_api(texts)
+    def __call__(self, texts: List[str], task_type: Optional[str] = None) -> List[np.array]:
+        return self._call_jina_api(texts, task_type=task_type)
 
-    def _call_jina_api(self, texts: List[str]):
+    def _call_jina_api(self, texts: List[str], task_type: Optional[str] = None):
+        data = {
+            "input": texts,
+            "model": self.model_name,
+            "dimensions": self.dimensions,
```
Comment (JoanFM) [inline milvus_model/dense/jinaai.py:59]: please check that `self.dimensions` and `task_type` are not None before putting them in the `data`

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Comment by wxywb** [comment]
The current design puts `task_type` in the __init__ method, and the __call__ method will use this setting. If someone needs to change it, they can create a new embedding function.
You can see an example in cohere.
https://github.com/milvus-io/milvus-model/blob/main/milvus_model/dense/cohere.py

**Pull request closed without merging**